test/python/WMComponent_t/WorkflowUpdater_t/WorkflowUpdater_t.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+import os
+import threading
+import unittest
+
+from WMCore_t.WMSpec_t.TestSpec import createTestWorkload
+from WMComponent.WorkflowUpdater.WorkflowUpdaterPoller import WorkflowUpdaterPoller
+from WMCore.Agent.HeartbeatAPI import HeartbeatAPI
+from WMCore.DAOFactory import DAOFactory
+from WMCore.DataStructs.Run import Run
+from WMCore.Services.UUIDLib import makeUUID
+from WMCore.WMBS.File import File
+from WMCore.WMBS.Fileset import Fileset
+from WMCore.WMBS.Subscription import Subscription
+from WMCore.WMBS.Workflow import Workflow
+from WMCore.WMSpec.Makers.TaskMaker import TaskMaker
+from WMQuality.Emulators import EmulatorSetup
+from WMQuality.Emulators.EmulatedUnitTestCase import EmulatedUnitTestCase
+from WMQuality.TestInitCouchApp import TestInitCouchApp as TestInit
+
+
+class WorkflowUpdaterTest(EmulatedUnitTestCase):
+    """
+    Test case for the WorkflowUpdater
+    """
+
+    def setUp(self):
+        """
+        Setup the database and logging connection. Try to create all of the
+        WMBS tables.
+        """
+        super(WorkflowUpdaterTest, self).setUp()
+
+        self.testInit = TestInit(__file__)
+        self.testInit.setLogging()
+        self.testInit.setDatabaseConnection(destroyAllDatabase=True)
+
+        self.testInit.setSchema(customModules=['WMCore.WMBS', 'WMCore.Agent.Database'],
+                                useDefault=False)
+
+        self.configFile = EmulatorSetup.setupWMAgentConfig()
+
+        myThread = threading.currentThread()
+        self.daoFactory = DAOFactory(package="WMCore.WMBS",
+                                     logger=myThread.logger,
+                                     dbinterface=myThread.dbi)
+        self.listActiveWflows = self.daoFactory(classname="Workflow.GetUnfinishedWorkflows")
+
+        self.testDir = self.testInit.generateWorkDir()
+        self.cwd = os.getcwd()
+
+        # Set heartbeat
+        self.componentName = 'WorkflowUpdater'
+        self.heartbeatAPI = HeartbeatAPI(self.componentName)
+        self.heartbeatAPI.registerComponent()
+
+        # don't execute tearDown because it hangs
+        # self._teardown = False
+
+        return
+
+    def tearDown(self):
+        """
+        Drop all of the non-sql and sql databases
+        """
+        self.testInit.clearDatabase(modules=['WMCore.WMBS', 'WMCore.Agent.Database'])
+        self.testInit.delWorkDir()
+        self.testInit.tearDownCouch()
+        EmulatorSetup.deleteConfig(self.configFile)
+
+    def getConfig(self):
+        """
+        Creates a common config.
+        """
+        config = self.testInit.getConfiguration()
+        self.testInit.generateWorkDir(config)
+
+        # First the general stuff
+        config.section_("General")
```

Comment:
I want to point out as a general comment here. Usage of custom WMCore configuration significantly impact code portability to other frameworks, like Flask, etc. There is no need to use it everywhere, and more standard format for configuration, like, `.ini`, `.json`, `.yaml` is a better choice. Having relying on custom format add extra layer of dependency without much of gain in code structure or functionality, and make it impossible to port (parts of) code to other languages which does not have WMCore python based configuration. Said that, there is nothing here to fix or address, and my comment is a general observation.